api/wallet_router_integration_test.go

```diff
@@ -0,0 +1,322 @@
+package api
+
+import (
+	"encoding/json"
+	"math/big"
+
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/google/uuid"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/multiaccounts"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/protocol/requests"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+	walletRequests "github.com/status-im/status-go/services/wallet/requests"
+	"github.com/status-im/status-go/services/wallet/responses"
+	"github.com/status-im/status-go/services/wallet/router"
+	"github.com/status-im/status-go/services/wallet/router/pathprocessor"
+	"github.com/status-im/status-go/services/wallet/router/sendtype"
+	"github.com/status-im/status-go/signal"
+	"github.com/status-im/status-go/t/utils"
+)
+
+const testAmount0Point1ETHInWei = 100000000000000000
+
+type routerSuggestedRoutesEnvelope struct {
+	Type   string                          `json:"type"`
+	Routes responses.RouterSuggestedRoutes `json:"event"`
+}
+
+func setupSignalHandler(t *testing.T) (chan responses.RouterSuggestedRoutes, func()) {
+	suggestedRoutesCh := make(chan responses.RouterSuggestedRoutes)
+	signalHandler := signal.MobileSignalHandler(func(data []byte) {
+		var envelope signal.Envelope
+		err := json.Unmarshal(data, &envelope)
+		assert.NoError(t, err)
+		if envelope.Type == string(signal.SuggestedRoutes) {
+			var response routerSuggestedRoutesEnvelope
+			err := json.Unmarshal(data, &response)
+			assert.NoError(t, err)
+
+			suggestedRoutesCh <- response.Routes
+		}
+	})
+	signal.SetMobileSignalHandler(signalHandler)
+
+	closeFn := func() {
+		close(suggestedRoutesCh)
+		signal.SetMobileSignalHandler(nil)
```

Comment:
Thanks for keeping an eye on the reset of `MobileSignalHandler` ❤️ 

I recently added a `ResetMobileSignalHandler`, please checkout [this](https://www.notion.so/Code-Smell-6-Cleanup-MobileSignalHandler-67dda79e7e0f4275b105af557d890b8d?pvs=4)

TLDR: Just add `t.Cleanup(signal.ResetMobileSignalHandler)` instead, it's a bit simpler.

Reply:
Updated.